db/savedata.go

```diff
@@ -142,3 +142,64 @@ func DeleteSessionSaveData(uuid []byte, slot int) error {
 
 	return nil
 }
+
+func RetrieveAccountEggs(uuid []byte) ([]defs.EggData, error) {
+	var accountEggs []defs.EggData
+
+	rows, err := handle.Query("SELECT uuid, gachaType, hatchWaves, timestamp FROM eggs WHERE owner = ?", uuid)
+	if err != nil {
+		return accountEggs, err
+	}
+
+	// For each row, we parse the raw data into an EggData and add it to the result
+	for rows.Next() {
+		var egg defs.EggData
+		err = rows.Scan(&egg.Id, &egg.GachaType, &egg.HatchWaves, &egg.Timestamp)
+		if err != nil {
+			return accountEggs, err
+		}
+
+		accountEggs = append(accountEggs, egg)
+	}
+
+	return accountEggs, nil
+}
+
+func UpdateAccountEggs(uuid []byte, eggs []defs.EggData) error {
+	tx, err := handle.Begin()
+	if err != nil {
+		return err
+	}
+
```

Comment:
```suggestion
defer tx.Rollback()
```

`tx.Rollback` is safe to call even after the `tx.Commit`